File: archive/count-items-matching-a-rule.js
const countMatches = function(items, ruleKey, ruleValue) {
  let ind
  if (ruleKey === 'type') {
    ind = 0
  } else
  if (ruleKey === 'color') {
    ind = 1
  } else {
    ind = 2
  }
  let ans = 0
  items.forEach(str => {
    if (ruleValue === str[ind]) {
      ans++
    }
  })

  return ans
};

console.log(countMatches([["phone","blue","pixel"],["computer","silver","lenovo"],["phone","gold","iphone"]],  "color", "silver"))

File: go-archive/count-items-matching-a-rule.go
func countMatches(items [][]string, ruleKey string, ruleValue string) int {
    ans:=0; val:=0
    if (ruleKey == "color") {val = 1} else if ruleKey == "name" {val = 2}
    for i:=0; i < len(items); i++ {
        if items[i][val] == ruleValue {ans++}
    }

    return ans
}